link-grammar/count.c

```diff
@@ -318,10 +314,6 @@ static Count_bin do_count(fast_matcher_t *mchxt,
 
 				/* Now, we determine if (based on table only) we can see that
 				   the current range is not parsable. */
-				Lmatch = (le != NULL) && (d->left != NULL) &&
-				         do_match(le, d->left, lw, w);
-				Rmatch = (d->right != NULL) && (re != NULL) &&
-				         do_match(d->right, re, w, rw);
 
```

Comment:
interting. Just pulling this out of the `lnull_cnt` loop, that alone might account for a big part of the speedup.